tests/vmi_cloudinit_test.go

```diff
@@ -132,14 +132,25 @@ var _ = Describe("[rfe_id:151][crit:high][vendor:cnv-qe@redhat.com][level:compon
 	Describe("[rfe_id:151][crit:medium][vendor:cnv-qe@redhat.com][level:component]A new VirtualMachineInstance", func() {
 		Context("with cloudInitNoCloud userDataBase64 source", func() {
 			It("[test_id:1615]should have cloud-init data", func() {
-				userData := fmt.Sprintf("#!/bin/sh\n\necho '%s'\n", expectedUserData)
+				userData := fmt.Sprintf("#!/bin/sh\n\ntouch /%s\n", expectedUserData)
 
 				vmi := tests.NewRandomVMIWithEphemeralDiskAndUserdata(cd.ContainerDiskFor(cd.ContainerDiskCirros), userData)
-				LaunchVMI(vmi)
+
+				By("Run the VMI")
+				vmi = tests.RunVMIAndExpectLaunch(vmi, 60)
+
+				By("waiting until login appears")
+				libwait.WaitUntilVMIReady(vmi, console.LoginToCirros)
+
+				By("validating cloud-init disk is 4k aligned")
 				CheckCloudInitIsoSize(vmi, cloudinit.DataSourceNoCloud)
+
+				By("validating user-data script executed")
 				VerifyUserDataVMI(vmi, []expect.Batcher{
-					&expect.BSnd{S: "\n"},
-					&expect.BExp{R: expectedUserData},
+					&expect.BSnd{S: fmt.Sprintf("cat /%s\n", expectedUserData)},
+					&expect.BExp{R: console.PromptExpression},
+					&expect.BSnd{S: "echo $?\n"},
+					&expect.BExp{R: console.RetValue("0")},
 				}, time.Second*120)
```

Comment:
You could replace all of this with `console.RunCommand` which embeds most of the logic and is quicker to feedback in case the command fails.